### ISSUE TITLE
i want use this network to train my dataset， but i don't konw how to use this network.

### ISSUE THREAD

**Issue by ycccco** [MCG-NJU/MeMOTR#32]
Dear authors: 
i met some troubles. i tried to transform my own data to the type of the open dataset （MOT17）, maybe my own MOT17 dataset architecture was wrong，so when i used the code(generate by ./data/gen_mot17_gts.py), it didn't work, nothing was generate。 
    And i don't konw how to disposition the train scripts.  the readme show the Scripts on DanceTrack like: python -m torch.distributed.run --nproc_per_node=8 main.py --use-distributed --config-path ./configs/train_dancetrack.yaml --outputs-dir ./outputs/memotr_dancetrack/ --batch-size 1 --data-root <your data dir path>.   
   Thank you very much for your contribution in this area.

**Comment by HELLORPG** [comment]
May I ask what the format of your own data is? Our `gen_mot17_gts.py` is used for transforming the MOT17 format to a custom format that can be handled by our dataloader.

**Comment by ycccco** [comment]
> May I ask what the format of your own data is? Our is used for transforming the MOT17 format to a custom format that can be handled by our dataloader.`gen_mot17_gts.py`

Dear author:
Sure!  The format of my own data is like this:

![Image](https://github.com/user-attachments/assets/4420eaeb-ed7f-4ac6-80f9-ee1e8e3c68c0)

is it wrong?  and the content of the gt.txt is: frame_count, id, x, y, w, h, activeta, label-id, visibility ratio.
the content of the det.txt is: <frame>, <id>, <bb_left>, <bb_top>, <bb_width>, <bb_height>, <conf>.
These are the format of my own dataset.  i consults the format of the open dataset mot17 to transform my own video data to this format.
Thank you very much for taking the time to answer my questions

**Comment by ycccco** [comment]
> May I ask what the format of your own data is? Our `gen_mot17_gts.py` is used for transforming the MOT17 format to a custom format that can be handled by our dataloader.

Dear author：
the problem in the format of MOT17 was solved, but when i tried to train the model in my own dataset, the error was occur：

![Image](https://github.com/user-attachments/assets/08dcc949-8079-455c-b2e4-e9343890c960)
In my own dataset, it just one set of the target, so i don't have the crowdhuman dataset, and what can i do to avoid this error.  
Thank you very much for taking the time to answer my questions

**Comment by HELLORPG** [comment]
By default, if you use `mot17.py` as your dataset file, the Dataset class will prepare for both the MOT17 and CrowdHuman datasets. If you want only to use a single dataset, you may choose one of the following:

1. Use `dancetrack.py` as your dataset class.
2. Delete the CrowdHuman part from `mot17.py`.
3. Maybe you could try to set `USE_CROWDHUMAN` to `False`. But I'm not sure if this parameter is available.

**Comment by ycccco** [comment]
> By default, if you use `mot17.py` as your dataset file, the Dataset class will prepare for both the MOT17 and CrowdHuman datasets. If you want only to use a single dataset, you may choose one of the following:
> 
> 1. Use `dancetrack.py` as your dataset class.
> 2. Delete the CrowdHuman part from `mot17.py`.
> 3. Maybe you could try to set `USE_CROWDHUMAN` to `False`. But I'm not sure if this parameter is available.

Dear author：

Thank you for your suggestion， when i try to attempt the suggest one, then i made the format of my own dataset to DanceTrack. but when i try to train the dataset, in the terminal, i input this order:
python main.py --config-path ./configs/train_dancetrack.yaml --outputs-dir ./outputs/memotr_dancetrack/ --batch-size 1 --data-root E:\Database\Trackset --use-checkpoint
the the bug(error) was occur:

![Image](https://github.com/user-attachments/assets/48253c5c-e9d1-48a0-95f0-b8dd8dd9d6c9)
i don't know how to solve this problem.  wish you have a nice day. Thank you for your help again!

**Comment by HELLORPG** [comment]
This error seems to be caused by system issues. Have you tried to train on our provided datasets? I wonder if this issue arises in our default experiments.

**Comment by ycccco** [comment]
> This error seems to be caused by system issues. Have you tried to train on our provided datasets? I wonder if this issue arises in our default experiments.

Dear author：
    Sorry about that，i was not found the dataset you provided.  In last time, i was use my own video to made into the format of Dancetrack dataset to train, i am not sure about that  the dataset is meet your provided  this is the format of dataset:

![Image](https://github.com/user-attachments/assets/6084e5cd-fd3e-4be2-abf8-7c2ec35afa0a)

And next time i will use the open dataset Dancetrack to make the format in this model. Then i will recode the circumstance in the train of Dancetrack dataset.  Thank you in advance!

**Comment by HELLORPG** [comment]
I mean, how about directly training our model on DanceTrack on your PC? I wonder if the issue is caused by the custom dataset or Windows system.

**Comment by ycccco** [comment]
> I mean, how about directly training our model on DanceTrack on your PC? I wonder if the issue is caused by the custom dataset or Windows system.

Dear author：
        i followed your suggestion，use the DanceTrack dataset to train on my PC, the error occur again:

![Image](https://github.com/user-attachments/assets/99505abd-5144-4251-9d1b-e06e644f2d0e)
and this is the part of DanceTrack dataset:

![Image](https://github.com/user-attachments/assets/4a6f11f9-27e6-4549-bb4d-b1a0d8305830)
so, i really don't know how to deal this problem. Thanks for your help!

**Comment by HELLORPG** [comment]
Now I know this issue. It is caused by `default(lambda: ...)` in Windows. You could refer to [the relevant reply](https://github.com/MCG-NJU/MOTIP/issues/24#issuecomment-2251901395). Since the reference issue is discussed in Chinese, feel free to let me know if you can't read Chinese.

**Comment by ycccco** [comment]
> Now I know this issue. It is caused by `default(lambda: ...)` in Windows. You could refer to [the relevant reply](https://github.com/MCG-NJU/MOTIP/issues/24#issuecomment-2251901395). Since the reference issue is discussed in Chinese, feel free to let me know if you can't read Chinese.

Dear author：
        Thanks for your help， this error was solved by this way：

![Image](https://github.com/user-attachments/assets/0826b035-cafb-459e-965c-c3742e07c764)

And in the time of this train. a 'Numpy is not available' error occur, this problem can be solved by "conda install "numpy<2.0" -c conda-forge" ， means degrade the Numpy in your environment.

**Comment by HELLORPG** [comment]
I'm glad you solved the problem. Thank you also for the information provided. At present, I still don't know what caused this error, but downgrading the version does solve this issue.
